flow/record/fieldtypes/__init__.py

```diff
@@ -734,3 +735,78 @@ def __repr__(self) -> str:
                 quote = '"'
 
         return f"{quote}{s}{quote}"
+
+
+class command(FieldType):
+    executable: str = None
+    args: list[str] = None
+    _posix: bool
+
+    def __new__(cls, value: str):
+        if cls is not command:
+            return super().__new__(cls)
+
+        if not isinstance(value, str):
+            raise ValueError(f"Expected a value of type 'str' not {type(value)}")
+
+        # pre checking for windows like paths
+        # This checks for windows like starts of a path:
+        #   an '%' for an environment variable
+        #   r'\\' for a UNC path
+        #   the strip and check for ":" on the second line is for `<drive_letter>:`
+        windows = value.startswith((r"\\", "%")) or value.lstrip("\"'")[1] == ":"
+
+        if windows:
+            cls = windows_command
+        else:
+            cls = posix_command
+        return super().__new__(cls)
+
+    def __init__(self, value: str | tuple):
+        if isinstance(value, tuple):
+            self.executable, self.args = value
+            self.args = list(self.args)
+            return
+
+        self.executable, self.args = self._split_function(value)
+
+    def _split_function(self, value: str) -> tuple[str, list[str]]:
+        executable, *args = shlex.split(value, posix=self._posix)
+        executable = executable.strip("'\" ")
+
+        return executable, args
+
+    def _pack(self):
+        command_type = PATH_WINDOWS if isinstance(self, windows_command) else PATH_POSIX
+        return ((self.executable, self.args), command_type)
+
+    @classmethod
+    def _unpack(cls, data: tuple[tuple[str, list], int]) -> command:
+        _value, _type = data
+        if _type == PATH_WINDOWS:
+            return windows_command(_value)
+
+        return posix_command(_value)
+
+    def __repr__(self):
```

Comment:
```suggestion
    def __repr__(self) -> str:
```
Can you also move it under `__init__`.